public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -2,12 +2,21 @@ import { css } from '@emotion/css';
 import React from 'react';
 
 import { DataFrame, FieldMatcherID, getFrameDisplayName, PanelProps, SelectableValue } from '@grafana/data';
-import { PanelDataErrorView } from '@grafana/runtime';
+import { PanelDataErrorView, reportInteraction } from '@grafana/runtime';
 import { Select, Table, usePanelContext, useTheme2 } from '@grafana/ui';
 import { TableSortByFieldState } from '@grafana/ui/src/components/Table/types';
 
 import { Options } from './panelcfg.gen';
 
+export const INTERACTION_EVENT_NAME = 'table_panel_usage';
+export const INTERACTION_ITEM = {
```

Comment:
Would it make sense for this to be an enum?

Reply:
In a way, it would, to make sure they are immutable, but from what I'm gathering const objects (eventually with an `as const` cast for immutability) are preferred. Overall I don't think this is a big problem especially since it's used for tracking.